graphrag_sdk/ontology.py

```diff
@@ -103,12 +104,54 @@ def from_graph(graph: Graph):
             )
 
         return ontology
+    
+    @staticmethod
+    def from_kg_graph(graph: Graph, node_limit: int = 100,):
+        """
+        Constructs an Ontology object from a given Knowledge Graph.
+
+        This function queries the provided knowledge graph to extract:
+        1. Entities and their attributes.
+        2. Relationships between entities and their attributes.
+
+        Args:
+            graph (Graph): The graph object representing the knowledge graph.
+            node_limit (int): The maximum number of nodes to sample for attribute extraction.
 
+        Returns:
+            Ontology: The Ontology object constructed from the Knowledge Graph.
+        """
+        ontology = Ontology()
+
+        # Retrieve all entity labels and relationship types from the graph.
+        e_labels = graph.query("call db.labels()").result_set
+        r_labels = graph.query("call db.relationshipTypes()").result_set
+        
+        # Process each entity label to extract attributes, limited to the specified number of nodes
+        for label in e_labels:
+            attributes = graph.query(
+                f"""MATCH (a:{label[0]}) call {{ with a return [k in keys(a) | [k, typeof(a[k])]] as types }} 
+                WITH types limit {node_limit} unwind types as kt RETURN kt, count(1)""").result_set
+            ontology.add_entity(Entity(label[0], [Attribute(attr[0][0], attr[0][1]) for attr in attributes]))
```

Comment:
it is possible for the same attribute name to show up multiple times, under different types e.g. numeric / string / boolean...